libs/net/dal/Services/ContentService.cs

```diff
@@ -288,17 +288,119 @@ public override Content AddAndSave(Content entity)
+                var values = (dynamic)kvp.Value;
+                return new
+                {
+                    Previous = values.DatabaseValue,
+                    Current = values.CurrentValue
+                };
+            }
+        );
+
+        this.Logger.LogInformation("Field changes detected:\n" +
+            string.Join("\n", changes.Select(c =>
+                $"Field: {c.Key}\n" +
+                $"  - Previous: {c.Value.Previous}\n" +
+                $"  - Current:  {c.Value.Current}"
+            ))
+        );
+    }
+
+    /// <summary>
+    /// Update and save the content entity. Handles concurrency conflicts by providing detailed information about changes.
+    /// </summary>
+    /// <param name="entity">The content entity to update</param>
+    /// <returns>The updated content entity</returns>
+    /// <exception cref="NoContentException">Thrown when the entity does not exist</exception>
+    /// <exception cref="DbUpdateConcurrencyException">Thrown when a concurrency conflict is detected</exception>
     public override Content UpdateAndSave(Content entity)
     {
-        var original = FindById(entity.Id) ?? throw new NoContentException("Entity does not exist");
-        this.Context.UpdateContext(original, entity);
-        if (entity.GuaranteeUid() && original.Uid != entity.Uid) original.Uid = entity.Uid;
-        return base.UpdateAndSave(original);
+        try
+        {
+            var original = FindById(entity.Id) ?? throw new NoContentException("Entity does not exist");
+            this.Context.UpdateContext(original, entity);
+            if (entity.GuaranteeUid() && original.Uid != entity.Uid) original.Uid = entity.Uid;
+            return base.UpdateAndSave(original);
+        }
+        catch (DbUpdateConcurrencyException ex)
+        {
+            this.Logger.LogInformation("Concurrency conflict detected - ContentId: {ContentId}", entity.Id);
+
+            var entry = ex.Entries.Single();
+            var currentValues = entry.CurrentValues;
+            var databaseValues = entry.GetDatabaseValues();
+
+            if (databaseValues == null)
+            {
+                var errorMessage = "Unable to retrieve latest content. Please refresh the page and try again. If the issue persists, contact support.";
+                this.Logger.LogError(errorMessage);
+                throw new DbUpdateConcurrencyException(errorMessage);
+            }
+
+            // Get changed properties and log details
+            var changedProperties = GetChangedProperties(currentValues, databaseValues);
+            LogFieldChanges(changedProperties);
+
+            var changedFields = string.Join("', '", changedProperties.Keys);
+
+            // Throw user-friendly message with better formatting
+            var userMessage = $"Content has been modified by another user. Modified fields: '{changedFields}'. " +
+                            "Please save your changes, refresh page, and reapply your updates.";
```

Comment:
It's impossible for the user to save their changes before refreshing the page.  It will result in the same error.

A better user experience would be to inform the user which fields are different.  Then they could choose whether they want to override the values in the database with their own.  You just need to be careful here, because many users will just say override.  Ideally we would have a way to determine what was changed, rather than what is different.  For example if the first save changes fields A and B, and the second save changes fields B and C.  Technically we can automatically resolve changes to fields A and C, but need to ask the user what to do with field B because they both edited it.

Reply:
Ideally if there is no difference other than the version number, or the status, it should be able to merge and save correctly without returning an error.